src/Config/Validator/Deploy/DatabaseSplitConnection.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\MagentoCloud\Config\Validator\Deploy;
+
+use Magento\MagentoCloud\Config\Database\DbConfig;
+use Magento\MagentoCloud\Config\Stage\DeployInterface;
+use Magento\MagentoCloud\Config\Validator;
+use Magento\MagentoCloud\Config\Validator\ResultFactory;
+use Magento\MagentoCloud\Config\ValidatorInterface;
+
+/**
+ * Validates the split database connections in DATABASE_CONFIGURATION variable
+ */
+class DatabaseSplitConnection implements ValidatorInterface
+{
+    /**
+     * @var ResultFactory
+     */
+    private $resultFactory;
+
+    /**
+     * @var DeployInterface
+     */
+    private $stageConfig;
+
+    /**
+     * @param ResultFactory $resultFactory
+     * @param DeployInterface $stageConfig
+     */
+    public function __construct(
+        ResultFactory $resultFactory,
+        DeployInterface $stageConfig
+    ) {
+        $this->resultFactory = $resultFactory;
+        $this->stageConfig = $stageConfig;
+    }
+
+    /**
+     * @return Validator\ResultInterface
+     */
+    public function validate(): Validator\ResultInterface
+    {
+        $customConfig = $this->stageConfig->get(DeployInterface::VAR_DATABASE_CONFIGURATION);
+
+        $customSplitConnections = [];
+        foreach (DbConfig::CONNECTION_TYPES as $connectionType) {
+            foreach (DbConfig::SPLIT_CONNECTIONS as $splitConnectionName) {
+                if (!isset($customConfig[$connectionType][$splitConnectionName])) {
+                    continue;
+                }
+                $customSplitConnections[$connectionType][] = $splitConnectionName;
```

Comment:
it can be collected in the `if` with changing the condition. Seems like it will be more readable.

Reply:
Done